docs/Developers/Profile_Bears.rst

```diff
@@ -0,0 +1,74 @@
+Profile Bears
+=============
+
+This document provides an overview of coala's Profiling Interface. The
+Profiling Interface provides abilities to profile Bear code to optimize its
+performance.
+
+The profiler will start by profiling the run() method of bears because this is
+the part where bear writers will spend time on, as rest of the part like loading
+the files, collecting the settings, etc. are done by coala itself.
+
+.. note::
+
+    Enabling both profiler and debugger on a bear at the same time will result
+    in error.
+
+Bear writers will have the ability to directly dump the raw profile output
+either on current working directory or to a specified directory name, which can
+be further used for examination of profiler stats with the help of different
+modules like ``pstats`` or ``snakeviz``.
+
+Usage
+-----
+
+coala's Profiler accepts an additional parameter, directory name or directory
+path where profiled files will be dumped, if ``True`` is provided profiled
+files will be saved to current working directory. If the specified directory
+does not exist it will be created. If the specified path points to an already
+existing file a ``FileExistsError`` is thrown.
```

Comment:
question is what that means to the regular user, but let's put that aside for now, I'm not sure how coala should handle such "user-exceptions" properly (which are intended to generate an error-log and exit).